File: 作业8-从一个有序数列中找出k个最接近x的元素/java/Solution.java
public class Solution {
	/**
	 * https://leetcode-cn.com/problems/find-k-closest-elements/solution/pai-chu-fa-shuang-zhi-zhen-er-fen-fa-python-dai-ma/
	 * 
	 * @param  nums [description]
	 * @param  k    [description]
	 * @param  x    [description]
	 * @return      [description]
	 */
	public static List<Integer> findClosestElements(int[] nums, int k, int x) {
		int size = nums.length;
		int left = 0;
		int right = size - 1;
		int removeLeft = size - k;
		while (removeLeft > 0) {
			if (x - nums[left] <= nums[right] - x) {
				right--;
			} else {
				left++;
			}
			removeLeft--;
		}
		List<Integer> ret = new ArrayList<>();
		for (int i = left; i < left + k; i++) {
			ret.add(nums[i])
		}
		return ret;
	}
}

File: 作业8-从一个有序数列中找出k个最接近x的元素/php/Solution.php
<?php

class Solution{
	function findClosestElement($nums, $k, $x) {
		$size = count($nums);
		$left = 0;
		$right = $size - 1;
		$removeLeft = $size - $k;
		while ($removeLeft > 0) {
			if ($x - $nums[$left] > $nums[$right] - $x) {
				$left++;
			} else {
				$right--;
			}
			$removeLeft--;
		}
		$map = [];
		for ($i = $left; $i < $left + $k; $i++) {
			$map[] = $nums[$i];
		}
		return $map;
	}
}

echo json_encode((new Solution())->findClosestElement([1,2,3,4,5], 4, 3)) . PHP_EOL;